### PR TITLE
Add MTL tutorial

### PR DESCRIPTION
**Pull request by vincentschen**
* Moved from snorkel/tutorials
* Fixed API signatures
* Update `split_data` using sklearn helpers

### CODE REVIEW

File: mtl/multitask_tutorial.py [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+# %% [markdown]
+# # Multi-Task Learning (MTL) Basics Tutorial
+
+# %% [markdown]
+# Multi-task learning is increasingly becoming a standard tool for the modern ML practioner. A major requirement of this regime is the ability to easily *add or remove new datasets, new label sets, new tasks, and new metrics*. Thus, in the Snorkel multi-task model design, each of these concepts have been decoupled.
+
+# %% [markdown]
+# The purpose of this tutorial is to introduce the basic classes and flow of the multi-task learning tools within Snorkel (not necessarily to motivate or explain multi-task learning at large; we assume prior experience with MTL).
+#
+# In this notebook, we'll look at a simple MTL model with only two tasks, each having distinct data and only one set of labels (the ground truth or "gold" labels). We'll also use data where the raw data is directly usable as features, for simplicity (i.e., unlike text data, where we would first need to tokenize and transform the data into token ids).
```
[the remark this answers is not in the document]
Comment (vincentschen) [reply mtl/multitask_tutorial.py]: 👍

File: mtl/multitask_tutorial.py [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+# ## Define Model
+
+# %% [markdown]
+# Now we'll define the `SnorkelClassifier`, which is build from a list of `Tasks`.
+
+# %% [markdown]
+# ### Tasks
+
+# %% [markdown]
+# A `Task` represents a path through a network. In Snorkel, this corresponds to a particular sequence of PyTorch modules that each `DataPoint` will pass through.
```
[the remark this answers is not in the document]
Comment (vincentschen) [reply mtl/multitask_tutorial.py]: 👍

File: mtl/multitask_tutorial.py [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+from functools import partial
+
+from snorkel.classification.task import Task, ce_loss, softmax
+from snorkel.classification.scorer import Scorer
+
+circle_task = Task(
+    name="circle_task",
+    module_pool=module_pool,
+    task_flow=task_flow,
+    loss_func=partial(ce_loss, "head_op"),
```
[the remark this answers is not in the document]
Comment (vincentschen) [reply mtl/multitask_tutorial.py]: It's explained once above: https://github.com/snorkel-team/snorkel-tutorials/blob/ebca8de0e980a362a0626384b9fde745fda7e333/mtl/multitask_tutorial.py#L196

File: mtl/multitask_tutorial.py [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+# %%
+from snorkel.classification.snorkel_classifier import SnorkelClassifier
+
+model = SnorkelClassifier([circle_task, square_task])
+
+# %% [markdown]
+# ### Train Model
+
+# %% [markdown]
+# Once the model is constructed, we can train it as we would a single-task model, using the `train_model` method of a `Trainer` object. The `Trainer` supports multiple schedules or patterns for sampling from different dataloaders; the default is to randomly sample from them proportional to their number of batches, such that all `DataPoints` will be seen exactly once before any are seen twice.
```
[the remark this answers is not in the document]
Comment (vincentschen) [reply mtl/multitask_tutorial.py]: feels like overkill? we can also call it random sampling, haha (cc @bhancock8)

File: mtl/multitask_tutorial.py [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+
+# "From the input dictionary, pull out 'circle_data' and send it through input_module"
+op1 = Operation(
+    name="input_op", module_name="input_module", inputs=[("_input_", "circle_data")]
+)
+
+# "From the output of op1 (the input op), pull out the 0th indexed output
+# (i.e., the only output) and send it through the head_module"
+op2 = Operation(
+    name="head_op", module_name="circle_head_module", inputs=[("input_op", 0)]
```
[the remark this answers is not in the document]
Comment (vincentschen) [reply mtl/multitask_tutorial.py]: what do you mean?

**Comment by vincentschen** [comment]
@henryre ready for another look!

File: mtl/multitask_tutorial.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (henryre) [reply mtl/multitask_tutorial.py]: Hmm not seeing anything specifically about the partial or the special loss methods

File: mtl/multitask_tutorial.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (henryre) [reply mtl/multitask_tutorial.py]: the second op below is called "square_head"

[the remark this answers is not in the document]
Comment (vincentschen) [reply mtl/multitask_tutorial.py]: got it—good catch!

File: mtl/multitask_tutorial.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (vincentschen) [reply mtl/multitask_tutorial.py]: you're right—adding a line!

**Pull request merged**